packages/rehype-mathjax/renderer.js

```diff
@@ -0,0 +1,25 @@
+const {mathjax} = require('mathjax-full/js/mathjax')
+const {TeX} = require('mathjax-full/js/input/tex')
+const {SVG} = require('mathjax-full/js/output/svg')
+const {liteAdaptor} = require('mathjax-full/js/adaptors/liteAdaptor')
+const {
+  RegisterHTMLHandler: registerHTMLHandler
+} = require('mathjax-full/js/handlers/html')
+const {AllPackages} = require('mathjax-full/js/input/tex/AllPackages')
+const adaptor = liteAdaptor()
+registerHTMLHandler(adaptor)
+const tex = new TeX({packages: AllPackages})
+const svg = new SVG({fontCache: 'none'})
+const html = mathjax.document('', {InputJax: tex, OutputJax: svg})
+
+module.exports = (math, display) => {
+  const css = adaptor.textContent(svg.styleSheet(html))
+  const pic = adaptor.outerHTML(html.convert(math, {display}))
+  return `
+    <span>${math}</span>
+    <script>
+    document.currentScript.previousElementSibling.attachShadow({mode:'open'});
+    document.currentScript.previousElementSibling.shadowRoot.innerHTML='<style>${css}</style>${pic}';
+    </script>
```

Comment:
What’s the reason for using a shadow root, if we have CSS and SVG? Can’t we embed those directly without scripts?

Reply:
Because MathJax has no all-in-one stylesheet like `katex.css`, we should generate stylesheet for each formula in my sense, we prefer to use ShadowDOM to avoid conflict styleshet of other formulae. As CHTML output requires webfont but `@font-face` property does not work in ShadowDOM (see also https://github.com/mdn/interactive-examples/issues/887), we use SVG output.

Reply:
>  we should generate stylesheet for each formula in my sense,

Excuse me, I misunderstood about stylesheet. The stylesheet is generated dynamically by MathJax *independently* to formulae. Thus, it is enough to be added at once. I'd like to add hook to append stylesheet after rendering all formulae if rehype provide hook for plugin. Is is it possible?

Reply:
Done.